### ISSUE TITLE
systemd-creds: Embedded credential name 'ciphertext.cred' does not match filename 'foobar', refusing.

### ISSUE THREAD

**Issue by cvlc12** [systemd/systemd#27970]
### systemd version the issue has been seen with

253.5-1

### Used distribution

Arch

### Linux kernel version used

6.3.6-arch1-1

### CPU architectures issue was seen on

None

### Component

systemd

### Expected behaviour you didn't see

```
systemd-creds encrypt plaintext.txt ciphertext.cred
shred -u plaintext.txt
systemd-run -P --wait -p LoadCredentialEncrypted=foobar:$(pwd)/ciphertext.cred systemd-creds cat foobar
```

should decrypt the credential

### Unexpected behaviour you saw

```
# systemd-creds encrypt plaintext.txt ciphertext.cred
# systemd-run -P --wait -p LoadCredentialEncrypted=foobar:$(pwd)/ciphertext.cred systemd-creds cat foobar
Running as unit: run-u151.service
Finished with result: exit-code
Main processes terminated with: code=exited/status=243
Service runtime: 506ms
CPU time consumed: 13ms
```
```
[root@bob ~]# systemctl status run-u151.service
× run-u151.service - /usr/bin/systemd-creds cat foobar
     Loaded: loaded (/run/systemd/transient/run-u151.service; transient)
  Transient: yes
     Active: failed (Result: exit-code) since Thu 2023-06-08 13:08:59 CEST; 22s ago
   Duration: 506ms
    Process: 5207 ExecStart=/usr/bin/systemd-creds cat foobar (code=exited, status=243/CREDENTIALS)
   Main PID: 5207 (code=exited, status=243/CREDENTIALS)
        CPU: 13ms

juin 08 13:08:58 bob systemd[1]: Started /usr/bin/systemd-creds cat foobar.
juin 08 13:08:59 bob (sd-mkdcre[5208]: Embedded credential name 'ciphertext.cred' does not match filename 'foobar', refusing.
juin 08 13:08:59 bob (md-creds)[5207]: run-u151.service: Failed to set up credentials: Protocol error
juin 08 13:08:59 bob (md-creds)[5207]: run-u151.service: Failed at step CREDENTIALS spawning /usr/bin/systemd-creds: Protocol error
juin 08 13:08:59 bob systemd[1]: run-u151.service: Main process exited, code=exited, status=243/CREDENTIALS
juin 08 13:08:59 bob systemd[1]: run-u151.service: Failed with result 'exit-code'.
```

On the other hand, when using LoadCredential (**not** encrypted):
```
# systemd-run -P --wait -p LoadCredential=foobar:$(pwd)/ciphertext.cred systemd-creds cat foobar
Running as unit: run-u152.service
k6iUCUh0RJCQyvL8k8q1UyAAAAABAAAADAAAABAAAACOG0XFQsvyorGcW7QAAAAAgAAAAAAAAAALACM
A8AAAACAAAAAAngAgqDZmTLqbUVqoOSwaIZPDHkmZyQF4uXLEL8AvQsNT0IgAEFDYPfewM1/IHzAImt
Mg5lclHmS0Enb8RMtFqpf0Vs1k4QPtzce/qX1bkb7xDpNe/xj+xcOjres9HRNMYWhrnLwtGRkcihZCN
0ZIRkZjuxc5Oe7J5AkSdPcMMRX5r7l37UagMyItq8WQUR6xiet4tda7GqUJtIIi/HvOAE4ACAALAAAA
EgAgY9RHOefBR+mMgV/Frr+UzQX1SIsLFItIbfB6Ep76It0AEAAgrOltCEx26E1UlzZfX1Or6jUUMB4
rIkBg/Rtz9gx5eFtj1Ec558FH6YyBX8Wuv5TNBfVIiwsUi0ht8HoSnvoi3QAAAACA/URmn64nVkxoga
S1sffjp7Qp+yKm5kL+3myhZaTx51X0vRYyADVKwqFtZMnwVCmWw28zREfNWa8O3neBTUByK+cICkBm
Finished with result: success
Main processes terminated with: code=exited/status=0
Service runtime: 40ms
CPU time consumed: 3ms
```

### Steps to reproduce the problem

Encrypting and decrypting work:

```
# echo -n hunter2 | systemd-creds encrypt - password.cred
# systemd-creds decrypt password.cred
hunter2
```

Am I doing something wrong ?

**Comment by poettering** [comment]
encrypted credentials carry the intended name in them, so they cannot be mispurposed. use "systemd-creds --name=" to pick the intended name in case yo don't want to derive it from the filename. You can also turn off this protection that way.

Please read the docs.

**Comment by poettering** [comment]
> Please read the docs.

Maybe *I* should read the docs from time to time ;-)

**Comment by poettering** [comment]
So this was already fixed actually, see 08b61b40ed5717c5186770b2a2f2a21816d7f6c1 aka #27387

But the website hasn't been updated. I am not sure what's going on there. @keszybz?

**Comment by poettering** [comment]
Closing this one as duplicate of #27387.

I have no idea how the website is now updated. Will inquire, but I guess we can close this issue here.
